test-sources/src/main/scala/simple_trees/SuperDoubleDiamond.scala

```diff
@@ -0,0 +1,30 @@
+package simple_trees
+
+object SuperDoubleDiamond:
+  // Lin = A, AnyRef
+  trait A:
+    def foo = 2
+
+  // Lin = B, AnyRef
+  trait B:
+    def foo = 3
+
+  // Lin = AA, A, AnyRef
+  trait AA extends A:
+    def foo: Int
+
+  // Lin = BB, B, AnyRef
+  trait BB extends B:
+    override def foo = super.foo
+
+  // Lin = AB, B, A, AnyRef
+  trait AB extends A, B:
+    override def foo = super.foo
+
+  // Lin = AAABBB, BB, AB, B, AA, A, AnyRef
+  class AAABBB extends AA, AB, BB:
+    override def foo = super.foo
+
+  // Lin = AABBAB, AB, BB, B, AA, A, AnyRef
+  class AABBAB extends AA, BB, AB
```

Comment:
Just for exhaustiveness I would put in the third possible permutation too, `ABAABB`.

Reply:
There are 3! = 6 permutations. Should I include them all?

Reply:
Well I was thinking that since `AA` and `BB` are positioned symmetrically in the inheritance graph, there are effectively only three permutations. But if you think the existing two permutations should already cover all situations then it’s fine.